From ec4121b1b093e26489a425b53a2e670239803058 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Wed, 24 Dec 2025 13:59:11 -0500
Subject: [PATCH 462/473] winewayland: Remove fractional scale redraw hack.

The issue lies somewhere else
---
 dlls/winewayland.drv/wayland_surface.c | 41 --------------------------
 1 file changed, 41 deletions(-)

diff --git a/dlls/winewayland.drv/wayland_surface.c b/dlls/winewayland.drv/wayland_surface.c
index 90aec786839..86d1a443ae1 100644
--- a/dlls/winewayland.drv/wayland_surface.c
+++ b/dlls/winewayland.drv/wayland_surface.c
@@ -138,46 +138,6 @@ static const struct xdg_toplevel_listener xdg_toplevel_listener =
     xdg_toplevel_handle_close
 };
 
-/*
-   needed to avoid race conditions with fractional scale redraw
-   TODO: move this
-*/
-static BOOL wayland_surface_refresh_contents(struct wayland_surface *wayland_surface)
-{
-    struct wayland_shm_buffer *shm_buffer;
-    struct wayland_win_data *data;
-    int width, height;
-    RECT *rect;
-    HRGN damage_region;
-    BOOL committed = FALSE;
-
-    if (!wayland_surface) goto done;
-    if (!(data = wayland_win_data_get_nolock(wayland_surface->hwnd))) goto done;
-    if (!(shm_buffer = data->window_contents)) goto done;
-
-    rect = &data->rects.window;
-    width = rect->right - rect->left;
-    height = rect->top - rect->bottom;
-
-    damage_region = NtGdiCreateRectRgn(0, 0, width, height);
-
-    if (wayland_surface_reconfigure(wayland_surface))
-    {
-        wayland_surface_attach_shm(wayland_surface, shm_buffer, damage_region);
-        wl_surface_commit(wayland_surface->wl_surface);
-        committed = TRUE;
-    }
-    else
-    {
-        TRACE("Wayland surface not configured yet, not flushing\n");
-    }
-
-    NtGdiDeleteObjectApp(damage_region);
-
-done:
-    return committed;
-}
-
 static void wp_fractional_scale_handle_scale(void *user_data,
         struct wp_fractional_scale_v1 *fractional_scale_v1, uint32_t scale)
 {
@@ -191,7 +151,6 @@ static void wp_fractional_scale_handle_scale(void *user_data,
         {
             surface->window.fractional_scale = scale / 120.0;
             surface->window.scale = surface->window.fractional_scale;
-            wayland_surface_refresh_contents(surface);
 
             TRACE("Got scale %lf\n", surface->window.fractional_scale);
         }
-- 
2.52.0

